perception/traffic_light_classifier/src/nodelet.cpp

```diff
@@ -98,19 +100,39 @@ void TrafficLightClassifierNodelet::imageRoiCallback(
   std::vector<cv::Mat> images;
   std::vector<size_t> backlight_indices;
   for (size_t i = 0; i < input_rois_msg->rois.size(); i++) {
-    output_msg.signals[i].traffic_light_id = input_rois_msg->rois.at(i).traffic_light_id;
+    // skip if the roi is not detected
+    if (input_rois_msg->rois.at(i).roi.height == 0) {
+      continue;
+    }
+    output_msg.signals[images.size()].traffic_light_id =
+      input_rois_msg->rois.at(i).traffic_light_id;
```

Comment:
[MUST]
In this approach, signal with height == 0 are removed. However, if this is not the last element in the vector, output_msg.signals[i] remains unassigned as the process continues.

Therefore, this function should be modified like:
```
void TrafficLightClassifierNodelet::imageRoiCallback(
  const sensor_msgs::msg::Image::ConstSharedPtr & input_image_msg,
  const tier4_perception_msgs::msg::TrafficLightRoiArray::ConstSharedPtr & input_rois_msg)
{
  if (classifier_ptr_.use_count() == 0) {
    return;
  }

  cv_bridge::CvImagePtr cv_ptr;
  try {
    cv_ptr = cv_bridge::toCvCopy(input_image_msg, sensor_msgs::image_encodings::RGB8);
  } catch (cv_bridge::Exception & e) {
    RCLCPP_ERROR(
      this->get_logger(), "Could not convert from '%s' to 'rgb8'.",
      input_image_msg->encoding.c_str());
  }

  tier4_perception_msgs::msg::TrafficSignalArray output_msg;

  std::vector<cv::Mat> images;

  for (size_t i = 0; i < input_rois_msg->rois.size(); i++) {
    // skip if the roi is not detected
    if (input_rois_msg->rois.at(i).roi.height == 0) {
      continue;
    }
    tier4_perception_msgs::msg::TrafficSignal tlr_sig;
    tlr_sig.traffic_light_id = input_rois_msg->rois.at(i).traffic_light_id;
    const sensor_msgs::msg::RegionOfInterest & roi = input_rois_msg->rois.at(i).roi;
    images.emplace_back(cv_ptr->image, cv::Rect(roi.x_offset, roi.y_offset, roi.width, roi.height));
    output_msg.signals.emplace_back(tlr_sig);
  }
  if (!classifier_ptr_->getTrafficSignals(images, output_msg)) {
    RCLCPP_ERROR(this->get_logger(), "failed classify image, abort callback");
    return;
  }

  // append the undetected rois as unknown
  for (const auto & input_roi : input_rois_msg->rois) {
    if (input_roi.roi.height == 0) {
      tier4_perception_msgs::msg::TrafficSignal tlr_sig;
      tlr_sig.traffic_light_id = input_roi.traffic_light_id;
      tier4_perception_msgs::msg::TrafficLightElement element;
      element.color = tier4_perception_msgs::msg::TrafficLightElement::UNKNOWN;
      element.shape = tier4_perception_msgs::msg::TrafficLightElement::CIRCLE;
      element.confidence = 0.0;
      tlr_sig.elements.push_back(element);
      output_msg.signals.push_back(tlr_sig);
    }
  }

  output_msg.header = input_image_msg->header;
  traffic_signal_array_pub_->publish(output_msg);
}
```